src/main/java/io/zipcoder/Classroom.java

```diff
@@ -48,15 +88,53 @@ public boolean addStudent(Student student) {
      * @param lastName
      * @return
      */
-    public Student removeStudent(String firstName, String lastName) {
-        return null;
+    public Student[] removeStudent(String firstName, String lastName) {
+        ArrayList<Student> postRemoval = new ArrayList<Student>();
+        Student[] postFormat = new Student[this.maxStudents];
+        for (int i = 0; i < this.studentsEnrolled; i++){
+            checkStudentEquivilenceAddToList(firstName, lastName, postRemoval, students[i]);
+        }
+        if (this.studentsEnrolled == postRemoval.size()) return null;
+        int index = 0;
+        createReturnArray(postRemoval, postFormat, index);
+        return postFormat;
+    }
+
+    private void createReturnArray(ArrayList<Student> postRemoval, Student[] postFormat, int index) {
+        for (Student keeper:postRemoval) {
+            postFormat[index] = keeper;
+            index++;
+        }
+    }
+
+    private void checkStudentEquivilenceAddToList(String firstName, String lastName, ArrayList<Student> postRemoval, Student student) {
+        if (!(student.getFirstName().equals(firstName) && student.getLastName().equals(lastName))){
+            postRemoval.add(student);
+        }
```

Comment:
this is all decent functional decomposition. it might be simpler though, once you've figured it all out, you could reduce the code by cleaning up the methods.